chore: beam 0.8 release notes (#203)
samply/beam: CHANGELOG.md

# Samply.Beam 0.8.0 - 2024-07-26

> This major release mostly features some bugfixes and the big upgrade of `hyper` to version 1.

Reviewer: ```suggestion
This major release of Beam 0.8 features many changes "under the hood", such as the highly anticipated upgrade of our `hyper` dependency to version 1, as well as many bug fixes. We were able to decrease the communication overhead between Beam.Proxies and the Beam.Broker and streamlined the behavior of some endpoints to make the usage of Samply.Beam simpler.
```

## Breaking changes

> * The `filter=todo` parameter on the tasks endpoint will now not yield tasks anymore which are claimed. This was bug but does change the behavior of the tasks endpoint in some cases.

Reviewer 2: ```suggestion
* The `filter=todo` parameter on the tasks endpoint will not yield tasks anymore that are claimed. This is a bugfix but changes the behavior of the tasks endpoint in some cases.
```

> * The `filter=todo` parameter on the tasks endpoint will now not yield tasks anymore which are claimed. This was bug but does change the behavior of the tasks endpoint in some cases.

Reviewer: ```suggestion
* The `filter=todo` parameter on the tasks endpoint will now not yield tasks anymore which are claimed. This change of behavior streamlines some often occurring usage patterns of this endpoint.
```

> * beam lib reexports types from `reqwest` which has been upgraded to the hyper 1 compatible version. This may lead to compile errors in code which uses old reqwest versions.

Reviewer 2: ```suggestion
* The library (`beam lib`) reexports types from `reqwest`, which has now been upgraded to use hyper 1.0. This leads to compile errors in code using previous reqwest versions.
```

> * beam lib reexports types from `reqwest` which has been upgraded to the hyper 1 compatible version. This may lead to compile errors in code which uses old reqwest versions.

Reviewer: ```suggestion
* beam lib reexports types from `reqwest` which has been upgraded to the hyper 1 compatible version. This may lead to compile errors in code bases using older reqwest versions.
```

## Minor changes

* Socket tasks may now hold additional metadata.
* New functions in beam lib to create sockets with metadata.

> * Added `beam_lib::RawString` a string wrapper to prevent beam lib from json encoding the string again. Useful for interacting with beam clients that don't use beam lib for serializing and deserializing types.

Reviewer 2: ```suggestion
* Added `beam_lib::RawString`, a string wrapper to prevent beam lib from json encoding the string again. Useful for interacting with beam clients that don't use beam lib for serializing and deserializing types.
```

> * Added `beam_lib::RawString` a string wrapper to prevent beam lib from json encoding the string again. Useful for interacting with beam clients that don't use beam lib for serializing and deserializing types.

Reviewer: ```suggestion
* Added `beam_lib::RawString`, a string wrapper to prevent beam lib from json encoding the string again. This is useful for beam_lib clients that handle serializing and deserializing types outside of beam_lib.
```

> * Added a `beam_lib::Error::InvalidReceivers` variant which holds all invalid receivers when trying to create a task with invalid recipients.

Reviewer 2: ```suggestion
* Added a `beam_lib::Error::InvalidReceivers` variant holding all invalid receivers when trying to create a task with invalid recipients.
```

> * Added a `beam_lib::Error::InvalidReceivers` variant which holds all invalid receivers when trying to create a task with invalid recipients.

Reviewer: ```suggestion
* Added a dedicated `beam_lib::Error::InvalidReceivers` error variant which holds all invalid receivers when trying to create a task with invalid recipients.
```

## Bugfixes

> * Fix broker deadlock when too many clients tried to update their certificates at the same time.

Reviewer 2: ```suggestion
* Fix broker deadlock when too many clients tried to fetch certificates at the same time.
```

> * Revalidate expired proxy certificates when they are resigned after an update.

Reviewer 2: ```suggestion
* Re-validate expired proxy certificates when they are re-signed after an update.
```

> * `filter=todo` does not yield claimed tasks anymore.

Reviewer 2: ```suggestion
```

This was already listed in the first section.

> * `filter=todo` does not yield claimed tasks anymore.

Reviewer: ```suggestion
```

> * Don't return already expired certificates to beam proxies.

Reviewer 2: ```suggestion
* The broker will now only return valid (non-expired) certificates to beam proxies.
```

> * Retry health check connection immediately after a gateway timeout.

Reviewer 2: ```suggestion
* beam-proxies will retry their health check connections to the broker immediately after a gateway timeout.
```
